### PR TITLE
Remove build dependency on `hg`/`mercurial`

### PR DESCRIPTION
**Pull request by alexeykazakov**
Just a cleanup. The same as https://github.com/fabric8-services/fabric8-wit/issues/2023

### CODE REVIEW

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-452-1`

File: .gitignore [hunk 1]
```diff
@@ -81,6 +81,3 @@ migration/sqlbindata_test.go
 configuration/confbindata.go
 wit/witservice
 account/tenant
-
-# Ignore artifacts directory
-bin/
```
Comment (sbose78) [inline .gitignore]: this should stay in the gitignore right? because the compiled artifacts are places here.

Comment (alexeykazakov) [reply .gitignore]: Sure. I'm just removing duplication - https://github.com/fabric8-services/fabric8-auth/blob/master/.gitignore#L68

**Comment by fabric8cd** [comment]
@alexeykazakov snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-452-2`

**Pull request merged**